documentation/token-structure.md

```diff
@@ -0,0 +1,430 @@
+Mag dat allebei?
+Blijkt hieruit dat we een level missen?
+Breek ik mijn eigen regels?)
+
+En toen dacht ik:
+
+1. Je hebt een kleur rood (`ams.color.red-60`) ← Brand Token
+2. Die noem je de error kleur (`ams.feedback.color.error`) ← ??? Token… Semantic?
+3. Je hebt een Text Input met een foute border (`ams.text-input.invalid.box-shadow`) ← Component Token
+4. Maar daar wil je eigenlijk een groep gebruiken (`ams.form-controls.border-color.invalid`) ← Common Token
+
+Hmmm…
+
+### Niveau wel of niet in naam token
+
+Dit is ook het moment om een knoop door te hakken of we al dan niet de niveaus ‘brand’ en ‘common’ expliciet in de namen van die tokens willen opnemen.
+
+Ze maken het superduidelijk op welk niveau je zit.
+Dus ook of gebruik van het token verplicht is (brand), of dat het je helpt om consistent te zijn (common).
+Het kan helpen bij autocomplete in de IDE: tik `ams.common.` om te zoeken of er al iets bestaat dat je kan gebruiken.
+Zo nee, probeer het nog eens met `ams.brand.`.
+
+Wel worden de namen natuurlijk iets langer.
+Misschien zijn er ander manieren om het onderscheid inzichtelijk te maken, of is dat niet per se nodig.
+
+Inhoudelijk zal niet altijd direct (en niet voor iedereen) duidelijk zijn of het tweede deel van de naam een property of een group is, zeker omdat de ‘property’ van brand ook iets anders mag zijn dan een CSS-property, zoals ‘space’.
+De vormen `ams.property.option` en `ams.group.property.option` zorgen al voor wat onderscheid tussen beide groepen omdat de ene altijd 3 elementen heeft en de andere 4.
+Maar Component Tokens kunnen er ook 3 of 4 hebben.
+Wel is het inhoudelijke onderscheid met Component Tokens iets duidelijker: de meeste mensen zullen `ams.button.*` en `ams.card.*` goed herkennen als Component Token, al is dat voor minder beroemde componenten niet altijd het geval.
+
+Het zou helpen als `group` altijd een meervoudsvorm zou zijn.
+Dan kun je zorgen voor: ‘ui-achtig ding’ --> component, ‘meervoudig woord’ --> common, ‘vormgevingsaspect’ --> brand.
+
+Zonder prefix zijn met name de Brand Tokens wel fantastisch kort.
+Iets als `ams.font-family.headings` zegt gewoon letterlijk ‘Dit is de Amsterdamse font family voor headings’, en `ams.font-size-md` net zo.
+Voor de kleuren is het wellicht nog de moeite waard om de `-60` eraf te poetsen zodat je `ams.color.blue` krijgt – hoe mooi wil je het hebben.
+
+Ook voor Common Tokens zijn bijvoorbeeld `ams.headings.color.default` en `ams.actions.cursor.disabled` best duidelijk.
+
+### Tokens aanroepen vanuit componenten
+
+Componenten mogen zowel Brand als Common Tokens aanroepen.
+Alleen Common Tokens toestaan is wat te strikt – in sommige gevallen moet je dan een letterlijke kopie maken van Brand naar Common.
+Ook NLDS geeft aan dat beide mag.
+
+Wel moeten mensen op de een of andere manier zorgen dat ze een Common Token kiezen als daar een toepasselijke van bestaat.
```

Comment:
Ik heb nu geen tijd om de hele PR door te nemen, maar hier zit volgens mij de crux van een issue met de voorgestelde opzet: Waarom publiceren we tokens die gebruikers niet mogen gebruiken?

Ik denk dat dit issue is ontstaan door deze keuze:

> [Een group is] Een gangbare term voor een groep componenten, óf voor een aspect van vormgeving of gedrag.

Als je 'een aspect van vormgeving of gedrag' in `brand` neerzet (bv `ams.brand.color.feedback.error`) en _group_ / _common_ bewaart voor groepen tokens die in meerdere componenten voor komen, heb je dat issue niet meer denk ik.

Reply:
Dank voor dit inzicht! Die ‘óf’ was al een ‘modelling smell’ :)